contracts/staking/ValidatorManager.sol

```diff
@@ -384,4 +371,59 @@ abstract contract ValidatorManager is
         ValidatorManagerStorage storage $ = _getValidatorManagerStorage();
         $._validationPeriods[validationID].weight = weight;
     }
+
+    /**
+     * @dev Helper function to check if the stake amount to be added exceeds churn thresholds.
+     */
+    function _checkAndUpdateChurnTrackerAddition(uint64 weight) internal {
+        _checkAndUpdateChurnTracker(weight, true);
+    }
+
+    /**
+     * @dev Helper function to check if the stake amount to be removed exceeds churn thresholds.
+     */
+    function _checkAndUpdateChurnTrackerRemoval(uint64 weight) internal {
+        _checkAndUpdateChurnTracker(weight, false);
+    }
+
+    /**
+     * @dev Helper function to check if the stake weight to be added or removed would exceed the maximum stake churn
+     * rate for the past churn period. If the churn rate is exceeded, the function will revert. If the churn rate is
+     * not exceeded, the function will update the churn tracker with the new weight.
+     */
+    function _checkAndUpdateChurnTracker(uint64 weight, bool addition) private {
+        ValidatorManagerStorage storage $ = _getValidatorManagerStorage();
+
+        if ($._maximumChurnPercentage == 0) {
+            return;
+        }
+
+        uint256 currentTime = block.timestamp;
+        ValidatorChurnPeriod memory churnTracker = $._churnTracker;
+
+        if (
```

Comment:
nit: Maybe we can add a comment to the branches of the conditional here, and maybe explain why the weight is always additive tot he `churnAmount` regardless of whether it's weight being added or removed. 

It is explained in the doc comment above so can drop it.